src/diffusers/pipelines/stable_diffusion/safety_checker.py

```diff
@@ -65,16 +69,20 @@ def forward(self, clip_input, images):
 
             result.append(result_img)
 
-        has_nsfw_concepts = [len(res["bad_concepts"]) > 0 for res in result]
+        has_nsfw_concepts = torch.tensor([len(res["bad_concepts"]) > 0 for res in result])
 
         for idx, has_nsfw_concept in enumerate(has_nsfw_concepts):
             if has_nsfw_concept:
-                images[idx] = np.zeros(images[idx].shape)  # black image
+                images[idx] = torch.zeros(images[idx].shape)  # black image
 
         if any(has_nsfw_concepts):
             logger.warning(
                 "Potential NSFW content was detected in one or more images. A black image will be returned instead."
                 " Try again with a different prompt and/or seed."
             )
 
+        if is_input_numpy:
+            images = images.numpy()
+            has_nsfw_concepts = has_nsfw_concepts.numpy()
+
```

Comment:
backward-compatible outputs of the safety checker (for community pipelines that use it)